src/spec-lang/expr_parser_header.ts

```diff
@@ -75,24 +76,83 @@ export type ExprParseOptions = {
     file: SourceFile,
     baseOff: number,
     baseLine: number,
-    baseCol: number
+    baseCol: number,
+    idxToOffMap: IdxToOffMap
 }
 
-function buildBinaryExpression(head: SNode, tail: Array<[string | undefined, BinaryOperator, string | undefined, SNode]>, src?: Range): SNode {
-    return tail.reduce((acc, [whiteSp, curOp, whiteSP, curVal]) =>
-        new SBinaryOperation(acc, curOp, curVal, src), head);
+function buildBinaryExpression(
+    head: SNode,
+    tail: Array<[BinaryOperator, SNode, PegsRange]>,
+    headRange: PegsRange,
+    opts: ExprParseOptions): SNode
+{
+    return tail.reduce((acc, [curOp, curVal, curLoc]) =>
+        new SBinaryOperation(acc, curOp, curVal, makeRange({start: headRange.start, end: curLoc.end})), head);
 }
 
-export function parseAnnotation(str: string, ctx: ASTNode, inference: InferType, file: SourceFile, baseOff: number): SAnnotation {
+const seen = new Set();
```

Comment:
I hope we would not get memory issues with this, as we never reset it. Maybe we should...

Reply:
Ooops, that was debugging only. Removed in [0d0da14](https://github.com/Consensys/scribble/pull/253/commits/0d0da1481ae42595676f591d33fb581891ae5846)